src/dmd/dsymbolsem.d

```diff
@@ -4397,15 +4397,38 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
             // Treat the remaining entries in baseclasses as interfaces
             // Check for errors, handle forward references
+            bool multiClassError = false;
+
             for (size_t i = (cldec.baseClass ? 1 : 0); i < cldec.baseclasses.dim;)
             {
                 BaseClass* b = (*cldec.baseclasses)[i];
                 Type tb = b.type.toBasetype();
                 TypeClass tc = (tb.ty == Tclass) ? cast(TypeClass)tb : null;
                 if (!tc || !tc.sym.isInterfaceDeclaration())
                 {
-                    if (b.type != Type.terror)
-                        cldec.error("base type must be `interface`, not `%s`", b.type.toChars());
+                    // It's a class
+                    if (tc)
+                    {
+                        if (!multiClassError)
+                        {
+                            error(cldec.loc,"`%s`: D does not support multiple class inheritance." ~
+                                  " Use multiple interface inheritance and/or composition.", cldec.toPrettyChars());
+                            multiClassError = true;
+                        }
+
+                        if (tc.sym.fields.dim)
+                            errorSupplemental(cldec.loc,"`%s` has fields: consider making it a member of `%s`",
```

Comment:
I believe this this should have a semicolon, not a colon "... has fields; consider..."

Reply:
A semicolon delimits a list, thats not really a list. Also there would no cooling preceding it which would be more weird. (Also also, when I'm programming semicolons mean something completely different ;))

Reply:
I agree with @JinShil  that a colon is not really appropriate here.

Reply:
A comma or a full stop then? 
A semicolon is much worse than a colon as the `errorSupplemental`s form items in a list. delimiting the propositions(?) _within_ the items with semicolons is just wrong.

Reply:
I think a comma would be fine.

Reply:
Done